src/vs/workbench/browser/parts/editor/editorGroupsControl.ts

```diff
@@ -760,7 +760,7 @@ export class EditorGroupsControl implements IEditorGroupsControl, IVerticalSashL
 			this.sashTwo.setOrientation(this.layoutVertically ? Orientation.VERTICAL : Orientation.HORIZONTAL);
 
 			// Trigger layout
-			this.arrangeGroups(GroupArrangement.EVEN);
+			this.arrangeGroups(GroupArrangement.KEEP_RATIO);
```

Comment:
Suggest to not introduce a `GroupArrangement.KEEP_RATIO` but rather allow to call `arrangeGroups` without any parameter and then use that as an indication to "keep the ratio". I would not expose this as API but just set the arrangement optional here in this file.